backend/src/main/java/com/soen341/instagram/controller/CommentController.java

```diff
@@ -0,0 +1,112 @@
+package com.soen341.instagram.controller;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.validation.Valid;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.dao.impl.PictureRepository;
+import com.soen341.instagram.dto.comment.CommentDTO;
+import com.soen341.instagram.dto.comment.CommentResponseDTO;
+import com.soen341.instagram.dto.picture.PictureDTO;
+import com.soen341.instagram.model.Comment;
+import com.soen341.instagram.model.Picture;
+import com.soen341.instagram.service.impl.CommentService;
+
+@RestController
+public class CommentController
+{
+	@Autowired
+	private CommentService commentService;
+	@Autowired
+	PictureRepository rep;
+	@Autowired
+	AccountRepository acrep;
+	@Autowired
+	private ModelMapper modelMapper;
+
+	@PostMapping(value = "/comment/newComment/{pictureId}")
```

Comment:
In REST-based API, we should serve data as if they were resources (perhaps like a file system). This would make more sense if the paths for these would be something like `/picture/{id}/comment/...`. Even if they are implemented as different entities in the backend, for the frontend it should just be abstracted as a resource.

Reply:
The refactoring can be done in next Sprint. This is ok for now

Reply:
I had the same idea too. But for me, I thought comment would be considered a ressource since we define it has an entity itself. Let me know if you think it's better to have it `picture/{id]/comment/...`